### PR TITLE
sdk-rust

### PR DESCRIPTION
**Pull request by talhahwahla**

### CODE REVIEW

**Comment by talhahwahla** [comment]
```
use mslm::Client;

#[tokio::main]
async fn main() {
    let c = Client::init("");
    let res = c.email_verify.single_verify("talha.hameed@mslm.io", None).await;

    match res {
        Ok(response) => {
            println!("Verification Response:\n{}", response);
        }

        Err(err) => {
            eprintln!("Error verifying email: {}", err);
        }
    }
}
```
```
$ cargo run --package mslm --example client
   Compiling mslm v1.0.0 (/home/talha/Desktop/sdk-rust)
    Finished dev [unoptimized + debuginfo] target(s) in 2.48s
     Running `target/debug/examples/client`
Verification Response:
{"email":"talha.hameed@mslm.io","username":"talha.hameed","domain":"mslm.io","malformed":false,"suggestion":"","status":"real","has_mailbox":true,"accept_all":false,"disposable":false,"free":false,"role":false,"mx":[{"host":"ASPMX.L.GOOGLE.COM.","pref":1},{"host":"ALT2.ASPMX.L.GOOGLE.COM.","pref":5},{"host":"ALT1.ASPMX.L.GOOGLE.COM.","pref":5},{"host":"ALT3.ASPMX.L.GOOGLE.COM.","pref":10},{"host":"ALT4.ASPMX.L.GOOGLE.COM.","pref":10}]}

```

**Comment by talhahwahla** [comment]
```shell
$ cat examples/otp_send.rs 
use mslm::Client;
use mslm::otp::OtpSendReq;

#[tokio::main]
async fn main() {
    let c = Client::init("<>");

    let otp_send_req = OtpSendReq {
        phone: String::from("<>"),
        tmpl_sms: String::from("Testing sdk-rust"),
        token_len: 6,
        expire_seconds: 60,
    };

    let res = c.otp.send(&otp_send_req, None).await;

    match res {
        Ok(response) => {
            println!("Response:\n{}", response);
        }

        Err(err) => {
            eprintln!("Error: {}", err);
        }
    }
}
```

```shell
$ cargo run --package mslm --example otp_send
   Compiling mslm v1.0.0 (/home/talha/Desktop/sdk-rust)
    Finished dev [unoptimized + debuginfo] target(s) in 2.54s
     Running `target/debug/examples/otp_send`
Response:
{"code":"1000","msg":"Successfully sent SMS."}
```

```shell
use mslm::Client;
use mslm::otp::OtpTokenVerifyReq;

#[tokio::main]
async fn main() {
    let c = Client::init("<>");

    let otp_token_verify_req = OtpTokenVerifyReq {
        phone: String::from("<>"),
        token: String::from("<>"),
        consume: Some(true),
    };

    let res = c.otp.verify(&otp_token_verify_req, None).await;
    match res {
        Ok(response) => {
            println!("Response:\n{}", response);
        }

        Err(err) => {
            eprintln!("Error: {}", err);
        }
    }
}
```

```shell
$ cargo run --package mslm --example otp_verify
   Compiling mslm v1.0.0 (/home/talha/Desktop/sdk-rust)
    Finished dev [unoptimized + debuginfo] target(s) in 2.48s
     Running `target/debug/examples/otp_verify`
Response:
{"code":"2001","msg":"Token is expired." }
```

File: src/otp.rs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    pub tmpl_sms: String,
+    pub token_len: i32,
+    pub expire_seconds: i32,
+}
+
+impl fmt::Display for OtpSendReq {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            r#"{{"phone":"{}","tmpl_sms":"{}","token_len":"{}","expire_seconds":"{}" }}"#,
```
Comment (UmanShahzad) [inline src/otp.rs:31]: Is there is no auto-derserialization of a struct into key/val pairs? Might be easier to maintain.

Same for all other `fmt::Display` impls

Comment (talhahwahla) [reply src/otp.rs:31]: Learned that using `#[derive(Serialize, Deserialize, Debug)]` attributes in conjunction with the `serde` automatically generates relevant implementations. So removing this `fmt::Display` code.

File: src/mslm.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+impl Client {
+    pub fn init(api_key: &'static str) -> Self {
+        let mut client = Client {
+            c: BaseClient::new(),
+            email_verify: EmailVerifyClient::new(api_key),
+            otp: OtpClient::new(api_key),
+        };
+        client.set_http_client(HttpClient::new());
+        unsafe {
```
Comment (UmanShahzad) [inline src/mslm.rs:32]: Is this Rust-Halal bro!?

**Review by UmanShahzad — APPROVED** [review]

File: src/mslm.rs (hunk 2 above)
Comment (talhahwahla) [reply src/mslm.rs:32]: From what is read [here](https://stackoverflow.com/questions/73419704/why-does-the-rust-standard-library-have-so-much-unsafe-code), it is. Will see if we can use better approach around this.

**Pull request merged**